packages/i18n-calypso/types/index.d.ts

```diff
@@ -0,0 +1,87 @@
+// Type definitions for i18n-calypso
+// Project: i18n-calypso
+
+/**
+ * External dependencies
+ */
+import { ComponentClass, ComponentType, ReactElement } from 'react';
+import moment from 'moment';
+
+declare namespace i18nCalypso {
+	export type Substitutions = string | string[] | { [placeholder: string]: string };
+
+	export interface ComponentInterpolations {
+		[placeholder: string]: ReactElement;
+	}
+
+	export interface TranslateOptions {
+		/**
+		 * Arguments you would pass into sprintf to be run against the text for string substitution.
+		 */
+		args?: Substitutions;
+
+		/**
+		 * Components for interpolation in the provided string.
+		 */
+		components?: ComponentInterpolations;
+
+		/**
+		 * Comment that will be shown to the translator for anything that may need to be explained about the translation.
+		 */
+		comment?: string;
+
+		/**
+		 * Provides the ability for the translator to provide a different translation for the same text in two locations (dependent on context). Usually context should only be used after a string has been discovered to require different translations. If you want to provide help on how to translate (which is highly appreciated!), please use a comment.
+		 */
+		context?: string;
+	}
+
+	export type TranslatedString = string;
```

Comment:
This probably adds no value because it appears as `string` for completions and typechecking for external consumers who are really the target audience.

Reply:
Removed alias